### PR TITLE
Feature/121 add separate discord block

### LINKED ISSUE #121: Add separate discord block
Add a block with links to telegram chats by location.
And maybe add a block with links to Discord channels of courses.

### PR DESCRIPTION
**Pull request by Tedzury**
## What type of PR is this? (select all that apply)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 🚧 Breaking Change
- [ ] 🧑‍💻 Code Refactor
- [ ] 📝 Documentation Update

## Related Tickets & Documents

- Related Issue #121 
- Closes #121 

## Screenshots, Recordings

Screenshot 2024-05-11 070625

## Added/updated tests?

- [x] 👌 Yes
- [ ] 🙅‍♂️ No, because they aren't needed
- [ ] 🙋‍♂️ No, because I need help

### CODE REVIEW

File: src/icons/discord-logo.tsx
Comment (SpaNb4) [inline src/icons/discord-logo.tsx:1]: Please refactor your code to follow the approach in this PR https://github.com/rolling-scopes/site/pull/191

File: src/features/communication/communication.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+export const Communication = ({ courseName }: RequiredProps) => {
+  return (
+    <section className="communication container">
+      <div className="communication content info-wrapper">
+        <Title text="Communication" hasAsterisk />
+        <div className="column-2">
+          <div className="disclogo-wrapper">
+            <DiscordLogo />
+          </div>
```
Comment (SpaNb4) [inline src/features/communication/communication.tsx:23]: I think this icon should be smaller, currently it is huge

File: src/features/communication/communication.scss [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        & .paragraph {
+          padding-top: 5px;
+        }
+
+        & .communication-link {
+          @include media-tablet {
+            font-size: 18px;
+          }
+
+          font-size: 22px;
```
Comment (ramzes-84) [inline src/features/communication/communication.scss:30]: As for me, the font size is too big

**Review by ramzes-84 — APPROVED** [review]

**Review by ramzes-84 — APPROVED** [review]

File: src/features/communication/communication.tsx (hunk 1 above)
Comment (Quiddlee) [inline src/features/communication/communication.tsx:23]: Can we get rid of some `div`'s

Comment (Quiddlee) [inline src/features/communication/communication.tsx:23]: `div` could be turned into `figure`

File: src/features/communication/communication.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            </Paragraph>
+            <Paragraph>
+              More info about communication at RS School in the{' '}
+              <a href={RS_DOCS_COMMUNICATION_LINK} target="_blank" className="communication-link">
+                RS Docs
+              </a>
+              . Here you could find rules, links to other courses and Telegram channels related to
+              your country or even city. Feel free to use it :)
+            </Paragraph>
+          </div>
```
Comment (Quiddlee) [inline src/features/communication/communication.tsx:41]: The wrapper `div` could be `article` instead

File: src/features/communication/communication.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+          <figure className="disclogo-wrapper">
+            <DiscordLogo />
+          </figure>
+          <div>
+            <Subtitle text="Discord is the main communication channel in RS School" />
+            <Paragraph>
+              Here is link for the{' '}
+              <a href={discordLinks[courseName]} target="_blank" className="communication-link">
+                course discord server
+              </a>{' '}
```
Comment (Quiddlee) [inline src/features/communication/communication.tsx:30]: Please make links to look like links, add underscore or external link icon, so user at the first look understand that this is a link and don't need to hover over them to realize that

image
or
image

Comment (SpaNb4) [reply src/features/communication/communication.tsx:30]: @Quiddlee, That's a great suggestion, but I'd say we should create a separate issue for that

File: src/features/communication/communication.tsx (hunk 1 above)
Comment (ansivgit) [reply src/features/communication/communication.tsx]: Maybe make the icon more colorful? Like app icon?
Screenshot 2024-05-13 at 17 23 21

File: src/shared/data/communication.data.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
+export const RS_DOCS_COMMUNICATION_LINK = 'https://docs.rs.school/#/rs-school-chats';
+
+export const discordLinks = {
```
Comment (ansivgit) [inline src/shared/data/communication.data.ts:3]:
```suggestion
export const DISCORD_LINKS = {
```

File: src/shared/data/communication.data.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  javascript: 'https://discord.com/invite/QvEYg7EaQ4',
+  'javascript-en': 'https://discord.com/invite/uW5cCHR',
```
Comment (ansivgit) [inline src/shared/data/communication.data.ts:5]: Let's rename it in `camelCase` style

File: src/features/communication/communication.tsx (hunk 4 above)
Comment (ansivgit) [reply src/features/communication/communication.tsx:30]: I suggest to create smth like `LinkExternal` component

File: src/shared/data/communication.data.ts
Comment (ansivgit) [inline src/shared/data/communication.data.ts:1]: Let's create folder `data` in `src` instead `shared/data`

File: src/features/about/about.data.ts
Comment (ansivgit) [inline src/features/about/about.data.ts:1]: Why do we need it so complicated?

File: src/shared/data/communication.data.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  react: 'https://discord.com/invite/zyRcphs3px',
+  'react ru': 'https://discord.com/invite/zyRcphs3px',
+  angular: 'https://discord.com/invite/xwReXYqvs7',
+  'node.js': 'https://discord.com/invite/8BFb8va',
+  'aws fundamentals': 'https://discord.com/invite/WEZxwRa4J6',
+  'aws cloud dev': 'https://discord.com/invite/WEZxwRa4J6',
+};
+
+export type CourseNames =
```
Comment (ansivgit) [inline src/shared/data/communication.data.ts:14]: Do we really need this type?

File: src/features/communication/communication.scss [hunk 8]
```diff
@@ -0,0 +1,56 @@
+.communication {
+  &.content {
+    &.info-wrapper {
+      display: flex;
+      flex-direction: column;
+      gap: 26px;
+
+      & .column-2 {
```
Comment (ansivgit) [inline src/features/communication/communication.scss:8]:
```suggestion
      .column-2 {
```
Please, fix here and below

File: src/shared/data/communication.data.ts (hunk 6 above)
Comment (Tedzury) [reply src/shared/data/communication.data.ts]: I suppose, we don't want to refactor half of the logic for courses at this branch, so I will leave it as is. The logic for about component, training programs and even course names spins around exactly these names and in need of refactoring.

File: src/features/about/about.data.ts
Comment (Tedzury) [reply src/features/about/about.data.ts:1]: I have no clue, it is not my invention)

File: src/shared/data/communication.data.ts (hunk 7 above)
Comment (Tedzury) [reply src/shared/data/communication.data.ts]: I found more elegant and generic way, in my opinion: 
```TS
export type CourseNames = keyof typeof DISCORD_LINKS;
```

**Review by Quiddlee — APPROVED** [review]

**Pull request merged**